lib/trento/infrastructure/software_updates/suma_api.ex

```diff
@@ -183,19 +294,24 @@ defmodule Trento.Infrastructure.SoftwareUpdates.SumaApi do
       |> String.split(";")
       |> Enum.find(&match_suma_session_cookie(&1))
 
-  defp extract_system_id(result) do
-    with false <- Enum.empty?(result),
-         %{id: system_id} <- Enum.at(result, 0) do
-      {:ok, system_id}
-    else
-      _ ->
-        Logger.error(
-          "Could not get system id for host from suma result. Result: #{inspect(result)}"
-        )
-
-        {:error, :system_id_not_found}
-    end
+  defp extract_system_id(
+         {:ok,
+          %{
+            success: true,
+            result: [
+              %{
+                id: system_id
+              }
+              | _
+            ]
+          }}
+       ) do
+    {:ok, system_id}
   end
 
+  defp extract_system_id({:ok, _}), do: {:error, :system_id_not_found}
+
+  defp extract_system_id({:error, _} = error), do: error
+
```

Comment:
It becomes way more visible what is expected. Love it, but it seems like you stripped the logging. Was that a conscious decision?

Reply:
You're actually right, it was absolutely unconscious, changing back :smile: